### PR TITLE
document merge options

### LINKED ISSUE #348: Document the Merge related CLI options
See: https://github.com/ConsenSys/teku/issues/4874

The options are currently hidden with the `-X` option, but we should probably document then as if there where visible (without `--X`), and merge the documentation as soon as the dev work is complete.

It would be good to have the doc ready when the feature is released.

### PR DESCRIPTION
**Pull request by alexandratran**
# Pull request checklist

Use the following template to make sure your PR fits the Teku documentation standard.

## Before creating the PR

Make sure that:

- [x] You've read the [contribution guidelines](https://consensys.net/docs/doctools/).
- [x] You've [previewed your changes locally](https://consensys.net/docs/doctools/en/latest/preview/old-system/#preview-locally).

## Describe the change

Document recently unhidden merge-related CLI options.

## Issue fixed

fixes #348 

## Impacted parts

For content changes:

- [x] Documentation content
- [ ] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (for example lint or vale)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [x] You've [fixed any issues](https://consensys.net/docs/doctools/en/latest/contribute/fix-cicd-errors/) raised by the tests.
- [x] You've [previewed your changes on Read the Docs](https://consensys.net/docs/doctools/en/latest/preview/old-system/#preview-on-read-the-docs)
  and added a [preview link](#preview).

## Preview

see https://pegasys-teku--349.com.readthedocs.build/en/349/Reference/CLI/CLI-Syntax/#ee-endpoint

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -2034,6 +2091,91 @@ When `LOGGING` is enabled, attestation and block performance is reported as log
 `METRICS` is enabled, attestation and block performance is reported using [metrics] in the
 [`VALIDATOR_PERFORMANCE`](#metrics-categories) metrics category.
 
+### validators-proposer-config
```
Comment (tbenr) [inline docs/Reference/CLI/CLI-Syntax.md:2094]: I think we should provide information about the json format.
the format is reported here:
https://github.com/ConsenSys/teku/pull/4894

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -2034,6 +2091,114 @@ When `LOGGING` is enabled, attestation and block performance is reported as log
...
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    validators-proposer-default-fee-recipient: "0xfe3b557e8fb62b89f4916b721be55ceb828dbd73"
+    ```
+
+Default fee recipient used if `fee_recipient` isn't specified in the
+[default proposer configuration](#validators-proposer-config), when producing a new execution block.
```
Comment (tbenr) [inline docs/Reference/CLI/CLI-Syntax.md:2200]:
```suggestion
When running a validator, this is an alternative to the [default proposer configuration](#validators-proposer-config) to specify a single fee recipient for all validator keys. 

When running a beacon node serving APIs to other validator clients, it is strongly suggested to use this parameter to specify fee recipient that will be used in rare cases when a validator client requests a block production but its fee recipient is still unknown for the beacon node.
```

This is a bit complicated and feel free to rephrase to make it more comprehensible

**Review by tbenr — APPROVED** [review]
LGTM with a suggestion. Feel free to improve wordings.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 3]
```diff
@@ -262,6 +262,63 @@ The default is `false`.
...
+=== "Example"
+
+    ```bash
+    --ee-endpoint=http://localhost:8545
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    TEKU_EE_ENDPOINT=http://localhost:8545
```
Comment (bgravenorst) [inline docs/Reference/CLI/CLI-Syntax.md:282]: I think the JSON-RPC port for the engine APIs differs to the default 8545, perhaps we should therefore use a different port here. Perhaps check what the ports are used in Besu for the `--engine...` CLI options.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 4]
```diff
@@ -262,6 +262,63 @@ The default is `false`.
...
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    ee-endpoint: "http://localhost:8545"
+    ```
+
+JSON-RPC URL of the execution engine node.
```
Comment (bgravenorst) [inline docs/Reference/CLI/CLI-Syntax.md:291]: Or perhaps something similar to, "URL of the execution client's Engine JSON RPC APIs."

Not sure about my wording here, but we should somehow draw a distinction between the standard Ethereum APIs and the Engine ones used by the consensus layer.

**Review by bgravenorst — COMMENTED** [review]
Left some feedback. There are a few gaps that we will need to be address when fully documenting the merge feature.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 3 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Syntax.md:282]: Or maybe I'm missunderstanding

Comment (alexandratran) [reply docs/Reference/CLI/CLI-Syntax.md:282]: fixed

File: docs/Reference/CLI/CLI-Syntax.md (hunk 4 above)
Comment (alexandratran) [reply docs/Reference/CLI/CLI-Syntax.md:291]: fixed

**Pull request merged**